src/javascripts/components/search.js

```diff
@@ -0,0 +1,82 @@
+/* global XMLHttpRequest */
+import accessibleAutocomplete from 'accessible-autocomplete'
+import lunr from 'lunr'
+
+var searchIndex = null
+var documentStore = null
+var timeout = 5 // Time to wait before giving up fetching the search index
+var search = {
+  fetchSearchIndex: function (callback) {
+    var request = new XMLHttpRequest()
+    request.open('GET', '/search-index.json', true)
+    request.timeout = timeout * 1000
+
+    request.onload = function () {
+      if (request.status >= 200 && request.status < 400) {
+        // Success!
+        var response = request.responseText
+        var json = JSON.parse(response)
+        callback(json)
+      } else {
+        // We reached our target server, but it returned an error
+        console.error('onload error', request.status, request.responseText)
+        // Log to analytics?
+      }
+    }
+
+    request.onerror = function (error) {
+      // There was a connection error of some sort
+      console.error('onerror', error)
+      // Log to analytics?
+    }
+
+    request.ontimeout = function (error) {
+      console.error('ontimeout', error)
+      // Log timeout to analytics?
+    }
+
+    request.send()
+  },
+  searchEngine: function (query, callback) {
+    var matchedResults = []
+    if (!searchIndex || !documentStore) {
+      return callback(matchedResults)
+    }
+    var searchResults = searchIndex.search(query + '*')
```

Comment:
Because this allows for all of lunr's syntax, things like fuzzy matching are allowed. This is powerful but dangerous, because it allows users to do things like create out of memory errors in their browser by attempting to search for `foo~100`.

It'd be safer for us to use `lunr.query` and set up our terms explicitly – unfortunately there's not much documentation on how to do that… but from [what docs there are](https://lunrjs.com/docs/lunr.Query.html) I _think_ it'd be something like…

```
var searchResults = searchIndex.query(function(q) {
  q.term(query, { wildcard: lunr.Query.wildcard.TRAILING })
})
```